src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/message/Split.java

```diff
@@ -0,0 +1,48 @@
+package uk.nhs.digital.nhsconnect.lab.results.model.edifact.message;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Split {
+    private static final char ESC_CHAR = '?';
+    private static final char SEGMENT_TERMINATOR = '\'';
+    private static final char FIELD_TERMINATOR = '+';
+    private static final char SUB_FIELD_TERMINATOR = ':';
+
+    public static String[] bySegmentTerminator(String input) {
+        return splitString(input, SEGMENT_TERMINATOR);
+    }
+
+    public static String[] byPlus(String input) {
+        return splitString(input, FIELD_TERMINATOR);
+    }
+
+    public static String[] byColon(String input) {
+        return splitString(input, SUB_FIELD_TERMINATOR);
+    }
+
+    private static String[] splitString(String input, char separator) {
```

Comment:
I guess this exists just as an optimisation, since `String::split` can only use a regex as the separator, which is more expensive?

Reply:
I think this was about handling escape chars. If I recall it was quite difficult to split taking escape char into account